### ISSUE TITLE
Keyboard navigation in content renderers

### ISSUE THREAD

**Issue by radinamatic** [learningequality/kolibri#8182]
## Observed behavior
When keyboard users open any resource, they should be able to navigate in and out the renderer frame, and interact with the content inside. This is valid for both _normal_ view and the fullscreen view.

After re-reading the specs and doing more research, I was not able to find much _normative_ requirements  regarding where should the focus be once the user tabs into an `iframe`, much less where it should be when the embedded content is in fullscreen. 

When talking about the [modal dialogs](https://www.w3.org/TR/wai-aria-practices/#dialog_modal), the recommendation is as follows:
> 1. When a dialog opens, focus placement depends on the nature and size of the content.
>    - In all circumstances, focus moves to an element contained in the dialog.
>    - Unless a condition where doing otherwise is advisable, focus is initially set on the first focusable element.
>    - If content is large enough that focusing the first interactive element could cause the beginning of content to scroll out of view, it is advisable to add `tabindex=-1` to a static element at the top of the dialog, such as the dialog title or first paragraph, and initially focus that element.
> ...

Comparison of embedded content in fullscreen with the modals comes to mind because the keyboard user is temporarily constrained to interact only with content that is inside the modal or frame in fullscreen. However, I've also found [recommendations to avoid trapping the focus](https://www.csun.edu/universal-design-center/web-accessibility-criteria-embedded-content#BP) inside the `iframe`. 

Regarding our existing content renderers, at this point I don't see problematic that, **once the fullscreen is activated, the focus remains on the *Exit fullscreen* button in the renderer topbar**, provided that:
1. The following <kbd>TAB</kbd> key presses enter the frame and follow the logical and visual order of the interactive elements inside.
2. All interactive elements inside the frame display a **clear visual indicator/outline when in focus** (we still seem to have some missing).

Chances are keyboard-only users know how to enter and exit the fullscreen, and will suffer more not knowing where they are if the focus visual indicator is missing. 

Below comments are valid for both _normal_ and the fullscreen view, and now I'm thinking I should open them as separate issues for each renderer.

Type | Comment
--- | ---
Video/Audio | Keyboard navigation functional, some focus visual indicators are missing (speed, CC, languages), and in general could benefit from a stronger outline (I forgot that back in a day I had to do the [same for KA Lite](https://github.com/learningequality/ka-lite/pull/4190)).
PDF | Impossible to navigate inside the frame with the keyboard: after the *View/Exit fullscreen* button, the following <kbd>TAB</kbd> key press focuses on the license details below the renderer. <kbd>+</kbd> and <kbd>-</kbd> buttons are missing the outline when in focus.
ePUB | Keyboard navigation functional, <kbd><</kbd> and <kbd>></kbd> buttons to navigate through the pages are often missing the outline when in focus.
HTML5 | - ASB: <kbd><</kbd> and <kbd>></kbd> buttons to navigate through the pages are not focusable, and the scrollbar at the bottom is missing the outline when in focus. <br> - Flexbook: OK<br> - PhET: not all sims are navigable by keyboard, but even in case of [those that are](https://phet.colorado.edu/en/simulation/molecules-and-light), it seems impossible to navigate inside the frame with the keyboard (after the *View/Exit fullscreen* button, the following <kbd>TAB</kbd> key press focuses on the license details below the renderer like for the PDFs).<br> - Blockly: Possible to navigate inside the frame, but content is not really navigable, drag & drop here is a massive challenge, not much we can do about.<br> - PLIX: Possible to navigate inside the frame, scroll the up and down, but otherwise the content is not navigable (what looks like a button is not really actionable with the keyboard 🙄) - again, not much we can do about.

As for the new custom navigation channels, it's hard to recommend anything specific without having more than our _super well designed and awesome_ one to look at... 🙂 

![Selection_004](https://user-images.githubusercontent.com/1457929/124089454-32d72280-da54-11eb-8538-59b3703ae1e3.png)

Is the topbar going to remain visible at all times? If yes, than we can consider the back arrow button as the equivalent of the *Exit fullscreen* button in our current renderers topbar, and the points 1 & 2 above should still apply. If not, then the creators of these channels should implement another way to exit the fullscreen and return to the list of channels in Learn. 

Ultimately, if the sighted keyboard-only user has a **clearly visible, _tabbable_ and actionable path to exit the fullscreen** (either with the permanently visible topbar, or a custom *Exit fullscreen* element inside the channel frame), it might not be a total anti-pattern to place the focus directly inside the `iframe` upon entering the fullscreen. For screen reader keyboard users benefit, we might consider providing a dedicated [keyboard shortcut](https://www.w3.org/WAI/WCAG21/quickref/#character-key-shortcuts) to exit fullscreen. 

# Errors and logs
…

## Expected behavior
Seamless access in & out of the content renderers, as well as interaction with the elements inside the `iframe`.
…

## User-facing consequences
Bad user experience for keyboard-only users.
…

## Steps to reproduce
…

## Context
 * Kolibri version: 0.14.7
 * Operating system: all
 * Browser: all

cc @rtibbles @jamalex

**Comment by marcellamaki** [comment]
@radinamatic -- just clarifying if this is specific to the _custom content renderer_ that we added. and @rtibbles -- do we have any idea of who might be using this so far? (trying to assess prioritization for the upcoming a11y patch)
